### ISSUE TITLE
Insufficient Funds while transferring 

### ISSUE THREAD

**Issue by Kindariegel** [DeFiCh/ain#913]
#### Actual behavior:

I can not send from the DEFI Wallet to another DFI address. 

#### Expected behavior:

I can send the DFI to another address w/o errors

#### How to reproduce it (as minimally and precisely as possible):

1. Screenshot of my UTXO and tokens:
![image](https://user-images.githubusercontent.com/41307159/141998763-b3e31d47-f632-4273-899d-a44aece31390.png)

2. Send to address brings up an error:
![image](https://user-images.githubusercontent.com/41307159/141999489-aff41a4c-fe38-4b32-b2fc-debaa73b6049.png)

3. Even with smaller amount, same error

4. CMD line:

![image](https://user-images.githubusercontent.com/41307159/141999849-71d65609-9802-456a-aac3-57fb9ffc965c.png)

#### What are your environment parameters?

- DeFiCh/app Version: 2.7.1.
- Operating System (OS): macOS 11.6
– RAM (GB): 32GB
– CPU (Intel Core i5/i7/etc.): i7
- Current free storage capacity: 300 GB free

#### Anything else we need to know?
[Archive.zip](https://github.com/DeFiCh/app/files/7546972/Archive.zip)

I started adding my issue to this bug https://github.com/DeFiCh/app/issues/1049 but I see different behaviours.

Thank you for supporting

**Comment by izzycsy** [comment]
Hi @Kindariegel , thanks for creating an issue, the team is currently investigating. Moving it to https://github.com/DeFiCh/ain for it to be tracked. Thank you.

**Comment by SilverSurfer2000** [comment]
A few days ago I tried to swap DFI to dBTC.

The (desktop) wallet converted the DFI to UXTO, but the following swap to dBTC did not work.

When trying to send or convert or do anything with those UXTO, I always get the error message 'insufficient funds'.

I found those UXTO via 'listaddressgroupings' on an unkonwn adress.

Those UXTO are now locked and I can't access them.

I already inserted the address manually to my wallet. With new DFI sent to this adress everything works ok (send, convert Token-UXTO, etc.).

However, the remaining UXTO from the swap 3 days ago remain locked and inaccessible. 

Therefore every transaction in the wallet with the 'max' button results in an 'insufficient funds' error - I need to manually insert an amount lesser these 'locked' UXTO.

**Comment by DerFuchs** [comment]
I've got the same problem, too. Tried different things like sending new UTXO and Token to that address, but nothing seems to work.

**Comment by SilverSurfer2000** [comment]
I was able to reproduce the malfunction.

I swapped DFI->dBTC.

The wallet converted all my funds to UXTO, but sent them to an internally newly generated adress.

After performing the swap to dBTC the remaining OXTO on the new adress show up in the wallet (and with cli getbalance), but are inaccesible.

Every transaction triggers the error code 'insufficient funds'.

I tried to send with GUI, cli sendtoaddress, accounttoutxos , utxostoaccount - same error.

When I send DFI to this new address, I can perform any send/trade up to the amount of these new dfi. 

Whenever the amount includes (parts 0f) the old (utxo-)dfi the error 'insufficient funds' is triggered again.

**Comment by cyrus1683** [comment]
Hi @izzycsy 

I also have the same problem. I can't send or convert my utxos to another address.
I created a vault and send 140 tokens to the vault, But now all my other DFI are also on the same owner address. (I didn't send them to that)
And from there I can't send it anyway out.
Even utxostoaccount and deposittovault doesn't work. Always the same error message "insufficient funds"

![Screenshot von 2021-11-18 08 54 46](https://user-images.githubusercontent.com/22840969/142374782-8d2710c0-5829-4cd2-9f00-910637d7e4c3.png)

getbalances
{
  "mine": {
    "trusted": 1942.73903334,
    "untrusted_pending": 0,
    "immature": 0
  }
}

listaccounts {} false false true
[
  {
    "key": "a9144f76294934e34cbd9a7a7346e26a9ba3a456470b87@167",
    "owner": "dLfSYvAgUTuj3j16WhdVhX68hKfBAn3hxE",
    "amount": "59.00000000@GOLD#167"
  },
  {
    "key": "a914716483a540650a45a6b9f702fc9827c6c1803f3587@167",
    "owner": "dPkrPJVSEfzZGfnyVpDNpsvwj8EiXPa16S",
    "amount": "59.00000000@GOLD#167"
  },
  {
    "key": "a914959cc8a4ee3e40f028aec0beff8e7929324575e987@0",
    "owner": "dT4N7tCMA26cyTw6vaKBiy7CgeW5MRz1wr",   # <--- Owneraddress of my vault
    "amount": "10.00000000@DFI"
  }
]

We talk about much money, so please fix that before the stocks go live!

Thanks and kind regards!
Daniel

**Comment by SilverSurfer2000** [comment]
Could we merge this topic with issue #1049, the last updates there refer to the same problem.

**Comment by SilverSurfer2000** [comment]
I found a workaround:

First you have to find out, on which address your UXTO are:
listaddressgroupings

Then use dumpprivkey in the CLI 
dumpprivkey ADDRESS

I installed the saiive.live desktop wallet and imported the private keys under Konten (arrow up top right corner).

Never give private keys to anyone or insert them in apps you don't know/trust!

In the saiive Wallet the locked UXTO could be sent and swapped without problems.

As the errer continues to happen in my main defi-wallet each time I try to send/swap UXTO I will now import my private keys into a new instance of the defi-wallet and repalce the wallet.dat (keep enough backups of the old wallet.dat!).
OP_DEFI_TX_AUTO_AUTH_PREP) and can't be accessed anymore. 

Maybe a timeout problem, maybe the wallet flags them incorrectly as locked...

**Comment by Kindariegel** [comment]
> I found a workaround:
> 
> First you have to find out, on which address your UXTO are: listaddressgroupings
> 
> Then use dumpprivkey in the CLI dumpprivkey ADDRESS
> 
> I installed the saiive.live desktop wallet and imported the private keys under Konten (arrow up top right corner).
> 
> Never give private keys to anyone or insert them in apps you don't know/trust!
> 
> In the saiive Wallet the locked UXTO could be sent and swapped without problems.
> 
> As the errer continues to happen in my main defi-wallet each time I try to send/swap UXTO I will now import my private keys into a new instance of the defi-wallet and repalce the wallet.dat (keep enough backups of the old wallet.dat!). OP_DEFI_TX_AUTO_AUTH_PREP) and can't be accessed anymore.
> 
> Maybe a timeout problem, maybe the wallet flags them incorrectly as locked...

Thanks @SilverSurfer2000 Your workaround works like a charm for me. 
I think the bug is still open and solved for DEFI Wallet itself

**Comment by prasannavl** [comment]
Hi all - thank you for reporting this.

**TL;DR**

- The **next release expected shortly will fix this**.
- Please **DO NOT use `dumpprivkey`** unless you have already done it, and know exactly what you are doing and trust the computing environment completely.
- More importantly, **please DO NOT PASTE any of these output** into any of these threads or anywhere else that you don't intend to. Once you lose control of it, there's no way to get it back. 

**Details**

The source of this issue was identified and the fix for this had also already been merged into the master branch with  https://github.com/DeFiCh/ain/pull/911. We do expect a release on top of the new 2.x very soon along with a number of quality of life fixes and improvements. We're are aiming to fast track the release to tomorrow, but should definitely be out in the few days.  

Meanwhile, I see some of the comments have noted dumping the wallet keys to move it to a different wallet. Just wanted to add a note that, **please do not use `dumpprivkey ADDRESS` unless you're aware of what it does, and the implications and nuances that come with exposing your raw private key.** While it's a small step, unless you're aware of each of the steps in-between, where it's copied, how it's copied, clipboard software, and completely trust all of the programs running in your system and the target wallet software, I'd recommend and appreciate your patience until we have the next release out.  

While the coin selection mechanism in the wallet caused this issue temporarily in some cases, the funds are unaffected and remain as they are. The next release should unblock this smoothly. However, if you chose to do the above private key dumping and if you lose control of your private keys in the process of transfer, please note that your funds may become irrecoverable.

**Comment by Kindariegel** [comment]
Thank you @prasannavl for bringing this notes up. 
I really appreciate your feedback regarding the upcoming release and fixing that issue

**Comment by dr-delphino** [comment]
Hello prasannavl,

Thank you for your input!

> Meanwhile, I see some of the comments have noted dumping the wallet keys to move it to a different wallet. Just wanted to add a note that, please do not use dumpprivkey ADDRESS unless you're aware of what it does, and the implications and nuances that come with exposing your raw private key. While it's a small step, unless you're aware of each of the steps in-between, where it's copied, how it's copied, clipboard software, and completely trust all of the programs running in your system and the target wallet software, I'd recommend and appreciate your patience until we have the next release out.

This is really interesting what you are addressing here. I'm really interested in learning more about this.
Is there a subtopic where these kind of issues are explained in more detail?

> While it's a small step, unless you're aware of each of the steps in-between, where it's copied, how it's copied, clipboard software, and completely trust all of the programs running in your system

What kind of implication can you derive from these facts?

I did use the cmd "dumpprivkey" on my own computer, but I didn't know that you should not  use it so lightly as you mentioned above.
There aro also many "how to tutorials" that suggest to save the private key as a "hard copy" on paper...(which I did)
I'm a little confused by your statement above and would like to know more about it.

**Comment by dr-delphino** [comment]
Hi there,

Can someone please help me or point me to an exisiting tutorial (if there is one- I couldn't find one that helped me so far).
I am currently having problems with liquidity mining on the Defi Chain Wallet.
It is not possible to add some liquidity (BTC - DFI) to a pool.

I tried again and again over the last few days, even with a lower amount of DFI.
But everytime I try, I get the error message "insufficient funds".


Even though I have ~4.766 DFI and ~ 0.00020906 dBTC.



- OS windows 10
- Defi chain wallet version 2.7.1

**Wallet Balance:**
![wallet](https://user-images.githubusercontent.com/58789151/142734830-1f33dbaa-27bc-49d9-8b46-03db8911149e.JPG)

**Error message:**
![insuffcient funds](https://user-images.githubusercontent.com/58789151/142734845-e7a1eebb-bb23-4e82-b581-b470a4978b22.jpg)

**I also took a screenshot of some CL output of my balances and accounts:**
![accounts](https://user-images.githubusercontent.com/58789151/142734889-5f355dc4-2d3e-47af-ac80-691941cfb831.jpg)
![accounts_2](https://user-images.githubusercontent.com/58789151/142734903-a0a72c98-7887-4bce-b16c-50994d6eb8dc.jpg)

**Comment by thedoublejay** [comment]
Hi all, `DeFi App v2.7.2` is updated with the fix for this issue. Please try it out and let us know if you're still facing this issue. Thanks - https://github.com/DeFiCh/app/releases/tag/v2.7.2
